### DISCUSSION TITLE
Additions!

### DISCUSSION THREAD

**Discussion by Dovidz** [marod424/tamarisk#11] [Ideas]
What additions do you want to add?

**Comment by Dovidz** [comment]
I want to add a dropdown that shows you all the assignments.

**Reply by 91-Jack** [reply]
We can make it color coded based on which units use similar syntax. Great idea Dovid!

**Reply by Aaronthetechguy** [reply]
UI ideas:
-Add better scoring page where you can see detailed scoring information 
-Add the ability to see what has previously been submitted to a given question
Backend idea:
-give information of what test failed
-have multiple correct scenarios for small changes
-more detailed error codes

**Comment by ebane01** [comment]
As the new and improved project manager, I would make many changes. First of all, I would make many changes that would include HTML code. I would center all of the text, videos and assignment descriptions because it makes the website look more professional. Secondly, I would write a series of assignments that would give students the ability to think about computer science in a more general sense, focusing on both the frontend and backend of coding. There would be four assignments that increasingly get more difficult but also focus on different aspects of coding:
The first two assignments would be focusing on frontside coding (HTML and JavaScript)
The first assignment would be taking a website someone else made and make it look prettier and more professional  
The second assignment would be building their own website with three pages: introduction, about them, something their passionate about and descriptions about why and what they do about it
The last two assignments would be focusing on the backside coding (Java):
The third assignment would be data inclusion. The assignment would provide different data to all the students and using that data, the student will have to write code to make the data in a specific order. For example, if the data was from a car dealership and it would be what cars were sold in a year, the code would sort the information by prices so the cars sold would be in an array of increasing order. 
This would allow students to think about using real world information and incorporating it into their code and how to write a code that has a specific job
The fourth assignment would be writing different methods for classes. This assignment will be straightforward coding. The point of this assignment is to get the students engaged with using getters, setters, and constructors in different scenarios and understanding the order of methods overriding one another. 
There is potential to make this assignment very difficult or quite similar. If the assignment is meant to be difficult, the instructions can say to include many different classes with many methods but if the assignment is meant to be easier, the students will write less methods and classes.

**Reply by marod424** [reply]
"As the new and improved project manager" 🤣 Love this

**Reply by marod424** [reply]
I really like your ideas around the focus of the assignments and the progression of their difficulty

**Comment by JLazarus12** [comment]
There is a need for more assignments as the AP Comp Sci class from the year 2021, finished all of the assignments way before the end of the school year. I would also add more assignments based on the AP curriculum and not just teaching, as most kids in the class are gonna take the AP. I would like to make a really modern and nice website with a rad logo. 
![TAMARISK LOGO ](https://user-images.githubusercontent.com/84030687/119866910-4ccc9500-beeb-11eb-8cd7-8680f0a89378.png)
![TAMARISK_free-file](https://user-images.githubusercontent.com/84030687/119866916-4e965880-beeb-11eb-8425-6d6c95f4919a.png)

**Reply by JLazarus12** [reply]
Also, I would love to add more detailed descriptions of what you errored in the code and more feedback. For example, when you write a code, and it is technically correct and works, but it's not the best way to write it, the website tells you the answer from a pre-made "answer key" showing the best way to write that code.

**Reply by marod424** [reply]
Love the designs, beautiful work!

**Comment by backjaum33** [comment]
This website truly has the potential to revolutionize the way computer science is taught in the classroom. There are a few optimizations that can be made:

Since someone signs in with their google account, we can use google tools to make the quality of life much better- calendar reminders for assignments, a shared leaderboard for assignments, etc.
Can also be designed to be more interactive- step-by-step instructions for coding, with a deep explanation for each line- should not replace the teacher but can really help to reinforce
The website should look better- a cool logo, clear font and directions, and convenient features- should make it enjoyable to code on

Overall, the website now is incomplete, but with these improvements, it can be made much better

**Comment by 91-Jack** [comment]
I want that you manually need to design a great picture using java code after every assignment starting with the assignment where we learned how to draw.

**Reply by marod424** [reply]
Maybe we can count this as an art class too!

**Comment by natem1839** [comment]
I would make many changes to Frisch Code Runner if I was the new project manager. I would make the website look a lot nicer. Not only is it not professional it is not easy to look at now. I would also edit all of the assignments so that you don't have to submit the whole code at the same time. I want the assignments to be submitted step by step to know where you need to improve. I want to have a video for every assignment so that there is not a problem of understanding what to do for the assignments. Also, I want to have a page for discussions on the site for people to ask questions and talk about the assignments. Maybe we can also have a calendar on the website to say when the assignment is due and possibly integrate with google calendar. After you finish the assignment and get a 100 it should be posted for other people who get an 100 should see in order to see the difference in their code. 
Thank you for reading!

**Comment by mcohen13** [comment]
Changes and Additional Feature Suggestions:

1. Have code recognize infinite loops and stop upon reaching them.
2. Highlight the lines of code with errors (bonus points if color coded by type i.e. red = syntax error, blue = logic error, yellow = infinite loop, etc.)
3. Auto-indent upon using keywords such as if, when, else, for, while, or a class/method opening curly bracket.
4. Be able to submit .java files. The old coderunner required us to copy + paste code that we wrote in an IDE, so why not just be able to upload the code as a file if it’s the same?
5.  A section with APCSA FRQ style coding assignments with multiple steps
6. Class forums to allow for collaboration and students to ask teachers and peers for help in a way that benefits the whole class
7. Email reminders and assignment due dates on the site (due dates could also have an effect on the grade for a given assignment, like -3% for every day overdue)
8. A style revamp, from instruction font to changing the colors of keywords

**Comment by marod424** [comment]
Love all of the ideas!

Sounds like the top priorities would be cleaning up the UI and better design and feedback around the assignments.
